### PR TITLE
Modernize the Jumpbox Genesis Kit

### PR DESCRIPTION
**Pull request by jhunt**
This PR modernizes the Jumpbox Genesis Kit so that it handles things a
bit more ergonomically, using new features available as of Genesis 2.6.0
and above.

This has been vetted manually in both vSphere and AWS environments.

### CODE REVIEW

**Review by davidlohle — APPROVED** [review]
LGTM! Tested on GCP and vSphere, docs are updated and manuals look good.

**Review by patjones — COMMENTED** [review]
Vetted on gcp and vsphere, docs look good. 👍

**Review by patjones — APPROVED** [review]

**Comment by jhunt** [comment]
I squash-merged offline, so that I could do the rebase in GH.

File: MANUAL.md [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+  This is a list of maps.
+
+  Each map represents a single user, and must contain the
+  following keys:
+
+  - `username` - The account name
+  - `shell` - Login shell, i.e. `/bin/bash` or `/bin/zsh`
+  - `ssh_keys` - A list of public SSH keys to authorize for
+    password-less authentication against this account.
+
```
Comment (dennisjbell) [inline MANUAL.md:42]: What happened to `env`?

File: MANUAL.md [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+```
+params:
+  jumbpox_network:   jumpbox
+  jumpbox_disk_pool: jumpbox # should be at least 50GB
+  jumpbox_vm_type:   jumpbox # VMs should have at least 1 CPU, and 2GB of memory
+```
+
+# Available Features
+
+- `azure` - Replaces BOSH's native "availability zones"
```
Comment (dennisjbell) [inline MANUAL.md]: azure should not be a feature, instead it should be detected in new, blueprint and check hooks

File: MANUAL.md [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+  - `renew-cert` - Renew the lifetime of an existing X.509 VPN
+    certificate, without changing the key that the user has.
+
+  - `reissue-cert` - Reissue an X.509 VPN certificate, and
+    generate a new key in the process.  This is useful if, for
+    example, a key has been lost or compromised.  The old
+    certificate will be revoked.
+
+  - `inventory` - Run the inventory errand against the deployment.
```
Comment (dennisjbell) [inline MANUAL.md]: `certs` is missing from here, but is a viable option in the list

File: hooks/addon [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+  echo "                them from accessing the VPN."
+  echo
+  echo "  renew-cert    Renew the lifetime of a previously-issued VPN"
+  echo "                certificate, without replacing the user's key."
+  echo
+  echo "  reissue-cert  Re-issue a VPN user certificate, regnerating the"
+  echo "                users key in the first place.  The old certificate"
+  echo "                will be revoked."
+  echo
+  echo "  inventory     Take an inventory of software installed on the jumpbox"
```
Comment (dennisjbell) [inline hooks/addon]: certs and reissue-all are not listed in list, but are in the case statement

File: hooks/blueprint [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+  shield)
+    echo >&2 "The Jumpbox Genesis Kit no longer supplies a 'shield' feature flag."
+    echo >&2 "If you wish to back up this jumpbox, please switch to using BOSH"
+    echo >&2 "runtime configurations to add the shield-agent to the deployment."
+    exit 1
+    ;;
+
+  proxy|azure)
+    echo >&2 "You no longer need to explicitly specify the 'proxy' feature."
```
Comment (dennisjbell) [inline hooks/blueprint]: replace 'proxy' with $want

File: hooks/info [hunk 6]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+set -eu
+ip=$(bosh vms --json | jq -r '.Tables[0].Rows[0].ips')
+
+describe "jumpbox ip: #C{$ip}"
```
Comment (dennisjbell) [inline hooks/info:5]: Not needed, but a list of users for a given environment would probably be a good idea.

**Review by dennisjbell — COMMENTED** [review]
hooks/new: cidr2mask doesn't check against negative numbers, but that's probably not crucial.  Also, potential name conflict of addons (addons already mean something in genesis) under manifest for openvpn feature, why not just call it manifest/features/openvpn.yml

File: MANUAL.md (hunk 2 above)
Comment (jhunt) [reply MANUAL.md]: azure is not even a thing, auto-detected or otherwise.  we don't HA the jumpboxen.

File: MANUAL.md (hunk 1 above)
Comment (jhunt) [reply MANUAL.md:42]: It's being phased out in favor of forthcoming docker image changes.  It's valid, but undocumented, on purpose.

File: hooks/blueprint (hunk 5 above)
Comment (jhunt) [reply hooks/blueprint]: Remove `azure` altogether; it's not a thing.

File: hooks/info (hunk 6 above)
Comment (jhunt) [reply hooks/info:5]: I believe the idea here was to remove the static_ips bits outright, and let the jumpboxen be dynamically IP'ed

Comment (dennisjbell) [reply hooks/info:5]: I was referring to the user list no being needed, not the jumpbox ip, and its definitely not a blocker to approval.

File: MANUAL.md (hunk 1 above)
Comment (dennisjbell) [reply MANUAL.md:42]: Premise accepted.

File: hooks/new [hunk 7]
```diff
@@ -42,8 +42,9 @@ trap "rm -f $GENESIS_ROOT/.$GENESIS_ENVIRONMENT.yml" INT QUIT TERM EXIT
 echo "kit:"
 echo "  name:    $GENESIS_KIT_NAME"
 echo "  version: $GENESIS_KIT_VERSION"
+echo "  features:"
+echo "    - (( replace ))"
```
Comment (jhunt) [inline hooks/new]: This isn't needed.

**Pull request merged**